fleetspeak_python/fleetspeak/server_connector/connector_test.py

```diff
@@ -15,105 +15,183 @@
 """Tests for grpcservice.client.client."""
 
 import threading
+import time
+from unittest import mock
 
 from absl.testing import absltest
 import grpc
+import grpc_testing
 
 from fleetspeak.server_connector import connector
 from fleetspeak.src.common.proto.fleetspeak import common_pb2
 from fleetspeak.src.server.proto.fleetspeak_server import admin_pb2
+from fleetspeak.src.server.proto.fleetspeak_server import admin_pb2_grpc
 
 
-# More of a mock than a fake.
-# TODO: migrate to use a mock library.
-class FakeStub(object):
+class RetryLoopTest(absltest.TestCase):
 
-  def __init__(self):
-    self.event = threading.Event()
+  @mock.patch.object(time, "sleep")
+  @mock.patch.object(time, "time", return_value=0)
+  def testNotSleepingOnFirstSuccessfulCall(self, time_mock, sleep_mock):
+    func = mock.Mock(return_value=42)
 
-    self.insert_done = False
-    self.insert_errors = 1
+    result = connector.RetryLoop(func, timeout=10.5, single_try_timeout=1)
 
-    self.delete_done = False
-    self.delete_errors = 1
+    func.assert_called_once()
+    sleep_mock.assert_not_called()
 
-    self.get_done = False
-    self.get_errors = 1
+    self.assertEqual(result, 42)
 
-    self.get_count_done = False
-    self.get_count_errors = 1
+  @mock.patch.object(time, "sleep")
+  @mock.patch.object(time, "time")
+  def testSingleTryTimeoutIsUsedForCalls(self, time_mock, sleep_mock):
+    cur_time = [0]
```

Comment:
Why a list and not just an integer (with `nonlocal` inside mocks).

Reply:
Fixed.